cmake/fmtlib.cmake

```
@@ -1,20 +1,7 @@
-include(FetchContent)
-
-FetchContent_Declare(
-	fmtlib
-	PREFIX "${PROJECT_BINARY_DIR}/deps"
-	DOWNLOAD_DIR "${PROJECT_SOURCE_DIR}/deps/downloads"
-	DOWNLOAD_NAME fmt-9.1.0.tar.gz
-	URL https://github.com/fmtlib/fmt/archive/9.1.0.tar.gz
-	URL_HASH SHA256=5dea48d1fcddc3ec571ce2058e13910a0d4a6bab4cc09a809d8b1dd1c88ae6f2
+include(${CMAKE_SOURCE_DIR}/cmake/submodules.cmake)
+set(FMT_SYSTEM_HEADERS ON)
+initialize_submodule(fmtlib)
```

super minor thing and feel free to ignore, but i think it would be clearer to set this stuff after the submodule has been initialized, that way it's clear that it really only pulls it in and doesn't really do anything else (same for the other deps)

```suggestion
initialize_submodule(fmtlib)
set(FMT_SYSTEM_HEADERS ON)
```

---

yep thats true